### PR TITLE
Make cs_args a known argument and handle c# external dependencies

### PR DESCRIPTION
**Pull request by thiblahute**
Avoiding warning about it when it is a well known one.

### CODE REVIEW

**Comment by thiblahute** [comment]
This makes our testsuite depend on glib-sharp-2.0 (not 3.0 as it is a mess pustream, I can change if you whish).

**Comment by thiblahute** [comment]
This fails because `glib-sharp-2.0` (ie. gtk-sharp-2.0) is not installed, how can I get it in there?

**Comment by jpakkane** [comment]
Add it to `ciimage/Dockerfile`. I will then update the CI image manually.

**Comment by thiblahute** [comment]
> Add it to ciimage/Dockerfile. I will then update the CI image manually.
Added.

**Comment by thiblahute** [comment]
> Add it to ciimage/Dockerfile. I will then update the CI image manually.

Added.

File: mesonbuild/backend/ninjabackend.py [hunk 1]
```diff
@@ -994,6 +994,11 @@ def generate_cs_target(self, target, outfile):
...
             deps.append(rel_src)
 
+        for dep in target.get_external_deps():
+            commands.extend_direct(dep.get_link_args())
+        commands += self.build.get_project_args(compiler, target.subproject)
+        commands += self.build.get_global_args(compiler)
+
         elem = NinjaBuildElement(self.all_outputs, outputs, 'cs_COMPILER', rel_srcs)
         elem.add_dep(deps)
         elem.add_item('ARGS', commands)
```
Comment (nirbheek) [inline mesonbuild/backend/ninjabackend.py]: This must be `commands.to_native()` if you use `CompilerArgs()`

**Pull request merged**